VideoLocker/src/main/java/org/edx/mobile/module/registration/view/IRegistrationFieldView.java

```diff
@@ -66,14 +66,8 @@ else if (fieldType.equals(RegistrationFieldType.MULTI)) {
                 return new RegistrationSelectView(field, view);
             }
             else if (fieldType.equals(RegistrationFieldType.CHECKBOX)) {
-                if (field.getAgreement() != null) {
```

Comment:
Do we really want to interpret all "checkbox" fields as a terms agreement? A better approach might be to either:
a) Update the API to include a property that will differentiate it from normal checkboxes, for example a different `type`, since we are not treating it like a checkbox at all.
b) Check if the field name == "honor_code" and only use the agreement logic for that.
c) Don't special case the agreement; treat all checkboxes the same way (this strategy would change the UX of course, because we aren't actually using a checkbox now)

Note that edx.org is currently using an actual checkbox. I'm not sure why the UI has diverged on mobile. Do we want to bring parity with the web here?